base/src/main/java/uk/ac/ebi/atlas/experimentimport/idf/IdfParserOutput.java

```diff
@@ -43,14 +44,21 @@ public int getExpectedClusters() {
     }
 
     public Set<String> getPubmedIds() {
-        return publications.stream().map(Publication::getPubmedId).collect(Collectors.toSet());
+        return publications.stream()
+                .map(Publication::getPubmedId)
+                .filter(Objects::nonNull)
+                .collect(Collectors.toSet());
     }
 
     public List<String> getMetadataFieldsOfInterest() {
         return metadataFieldsOfInterest;
     }
 
     public Set<String> getDois() {
-        return publications.stream().map(Publication::getDoi).collect(Collectors.toSet());
+        return publications
+                .stream()
+                .map(Publication::getDoi)
+                .filter(Objects::nonNull)
+                .collect(Collectors.toSet());
```

Comment:
This class has now some behaviour, so a small test would be in order.

Reply:
Will do.